api/pkg/apis/v1alpha1/model/validationrule.go

```diff
@@ -82,6 +82,9 @@ func detectChanges(properties []PropertyDesc, oldName string, newName string, ol
 			regexpPattern := strings.ReplaceAll(escapedPattern, `\*`, ".*")
 			// Compile the regular expression
 			regexpObject := regexp.MustCompile("^" + regexpPattern + "$")
+			if len(oldValues) != len(newValues) {
```

Comment:
I think this is over simplified. let's say the pattern is "a*"  and old value is ["ab"] while the new value is ["ab", "cd"]. Although the number of properties differ, "cd" is not matched by the pattern hence it should not be considered.

Reply:
Yes, this is not the final version. I am testing the behavior of skipIfMissing. After confirming the behavior, I will fix this PR and convert it to public.

Reply:
Fixed the logic. The compareProperties will work like this,

```go
func compareProperties(c PropertyDesc, old map[string]interface{}, new map[string]interface{}, key string) bool {
	v, ook := old[key]
	nv, nok := new[key]
	if c.PropChanged != nil {
		return c.PropChanged(v, nv)
	}

	if ook && nok {
		// case 1: key exists in both old and new
		// compare the values, if different, return true
		if !compareStrings(fmt.Sprintf("%v", v), fmt.Sprintf("%v", nv), c.IgnoreCase, c.PrefixMatch) {
			return true
		} else {
			return false
		}
	} else if !ook && !nok {
		// case 2: key does not exist in both old and new
		// return false to indicate no change
		return false
	} else {
		// case 3: one of them is missing
		// if the property is optional, no matter it doesn't exist in old or new, return false to indicate no change
		if !c.SkipIfMissing {
			return true
		} else {
			return false
		}
	}
}
```